### ISSUE TITLE
Feature: add the --builder flag support for build parameter for multi-arch builds

### ISSUE THREAD

**Issue by HarvyNBB** [docker/compose#10664]
### Description

* Brief:
Reference to the [documentation](https://docs.docker.com/compose/compose-file/build/)
Since it's possible to support multi-arch builds in docker-compose.yml, 

Additionally reference to the [documentation](https://docs.docker.com/build/architecture/)
```
In newer versions of Docker Desktop and Docker Engine, you’re using Buildx by default when you invoke the docker build command.
```

as a result, if the buildkit is enabled on the docker engine, then I assume docker compose build would use buildx

So during building simultaneous services with multi-arch there is a [buildkit issue](https://github.com/moby/buildkit/issues/3929) | [buildx issue](https://github.com/docker/buildx/issues/1868) faced while trying to build multi arch builds for multiple services

Note: I'm not sure whether this is buildx issue or buildkit issue, so I created an issue for both

so in order to get over this issue it would be so beneficial for multi-arch builds if the builder could be either passed via a flag or defined in the docker-compose.yml
for e.g docker-compose.yml

Multiple services:
```
version: '3.7'

services:
 
  service_1:
    container_name: service_1-${VERSION}
    image: service_1:${VERSION}    
    build:
      context: .
      dockerfile: Dockerfile
      builder: test1
      platforms: 
        - ${PLATFORM_1}
        - ${PLATFORM_2}
  
  service_2:
    container_name: service_2-${VERSION}
    image: service_2:${VERSION}
    build:
      context: .
      dockerfile: Dockerfile
      builder: test2
      platforms: 
        - ${PLATFORM_1}
        - ${PLATFORM_2}

  service_3:
    container_name: service_3-${VERSION}
    image: service_3:${VERSION}
    build:
      context: .
      dockerfile: Dockerfile
      builder: test3
      platforms: 
        - ${PLATFORM_1}
        - ${PLATFORM_2}
```

commands to run
```
docker buildx create --name test1
docker buildx create --name test2
docker buildx create --name test3
docker compose build service_1 service_2 service_3 
```
 Or for a single service
```
version: '3.7'

services:
 
  service_1:
    container_name: service_1-${VERSION}
    image: service_1:${VERSION}    
    build:
      context: .
      dockerfile: Dockerfile
      platforms: 
        - ${PLATFORM_1}
        - ${PLATFORM_2}
```

```
docker buildx create --name test1
docker compose build --builder test1 service_1 
```

**Comment by ndeloof** [comment]
We could indeed introduce `builder`, as a follow-up for https://github.com/compose-spec/compose-spec/issues/233

**Comment by milas** [comment]
Compose v2.20.0+ respects the `BUILDX_BUILDER` env var and also supports `build --builder=my-builder`.

Note that other commands that (might) perform builds like `up` and `run` do not support the `--builder` flag currently.

**Comment by jhrotko** [comment]
@glours @ndeloof any other work left in here?

**Comment by ndeloof** [comment]
@jhrotko compose-spec still does not support `builder`

**Comment by ndeloof** [comment]
Looking in more details into this issue, I notice this can be challenging to get a `builder` attribute _per_ service, as the docker/compose codebase run image builds in parallel on a single builder (according to `--builder`). Would need to fully redesign this code to support grouping services to build by builder then run builds ... while we agree the desired state is to fully delegate the build process to buildx (see https://github.com/docker/compose/pull/11484)
For this reason, I'm closing this as "not planned". If you consider this to still be a desirable feature, please create a follow-up issue and/or ask for support on buildx bake
